### PR TITLE
Added new workflow and debug messages

### PR DESCRIPTION
**Pull request by rbuzatu90**

### CODE REVIEW

File: HyperV/scripts/create-environment.ps1 [hunk 1]
```diff
@@ -4,6 +4,11 @@ Param(
     [string]$buildFor='openstack/compute-hyperv'
 )
 
+Write-Host "Debug info:"
```
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1]: Shouldn't this be done in debug case only?

File: HyperV/scripts/create-environment.ps1 [hunk 2]
```diff
@@ -157,8 +164,8 @@ if (Test-Path $pythonDir)
 }
 Write-Host "Ensure Python folder is up to date"
 Write-Host "Extracting archive.."
-& "C:\Program Files\7-Zip\7z.exe" x -y "$pythonArchive"
-& "C:\Program Files\7-Zip\7z.exe" x -y python27new.tar
+& $7zExec x -y "$pythonArchive"
+& $7zExec x -y "$pythonTar"
```
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1:171]: do we actually cleanup anywhere the .tar.gz and .tar file?

File: HyperV/scripts/create-environment.ps1 [hunk 3]
```diff
@@ -202,39 +209,55 @@ function cherry_pick($commit) {
     $ErrorActionPreference = $eapSet
 }
 
+Write-Host "BuildDir is: $buildDir"
```
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1]: Shouldn't this be only in debug mode?

File: HyperV/scripts/create-environment.ps1 [hunk 4]
```diff
@@ -202,39 +209,55 @@ function cherry_pick($commit) {
...
+Write-Host "ProjectName is: $projectName"
+Write-Host "Listing $buildDir parent directory:"
+Get-ChildItem ( Get-Item $buildDir ).Parent.FullName
+Write-Host "Listing $buildDir before install"
+Get-ChildItem $buildDir
+
 ExecRetry {
-    pushd C:\OpenStack\build\openstack\neutron
-    & pip install C:\OpenStack\build\openstack\neutron
+    Write-Host "Content of $buildDir\neutron"
```
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1]: where is that content?

File: HyperV/scripts/create-environment.ps1 [hunk 5]
```diff
@@ -202,39 +209,55 @@ function cherry_pick($commit) {
...
-    & pip install C:\OpenStack\build\openstack\compute-hyperv
+    Write-Host "Content of $buildDir\compute-hyperv"
+    Get-ChildItem $buildDir\compute-hyperv
+    pushd $buildDir\compute-hyperv
+    & pip install $buildDir\compute-hyperv    
     if ($LastExitCode) { Throw "Failed to install Hyperv-Compute fom repo" }
     popd
 }
 
 ExecRetry {
```
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1]: os-win setup should be before the others (as it's a library required by the other projects).

File: HyperV/scripts/create-environment.ps1 (hunk 2 above)
Comment (rbuzatu90) [inline HyperV/scripts/create-environment.ps1:171]: Yep, we do: 
if (Test-Path $pythonArchive)
{ Remove-Item -Force $pythonArchive}
and for tar also :)

File: HyperV/scripts/create-environment.ps1 (hunk 1 above)
Comment (rbuzatu90) [inline HyperV/scripts/create-environment.ps1]: Will remove that

File: HyperV/scripts/create-environment.ps1 (hunk 3 above)
Comment (rbuzatu90) [inline HyperV/scripts/create-environment.ps1]: I suggest to leave them, it wont hurt to have aditional info

File: HyperV/scripts/create-environment.ps1 (hunk 2 above)
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1:171]: :+1:

File: HyperV/scripts/create-environment.ps1 (hunk 1 above)
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1]: I was suggesting to put pines 7->10 inside an "if $debug"

File: HyperV/scripts/create-environment.ps1 (hunk 3 above)
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1]: same as above. We can leave it for the moment to test, but all these messages will have to be shown only for debug.

File: HyperV/scripts/create-environment.ps1 [hunk 6]
```diff
@@ -202,53 +212,71 @@ function cherry_pick($commit) {
...
+    }
+    if ($LastExitCode) { Throw "Failed to install os-win fom repo" }
     popd
 }
 
 ExecRetry {
-    pushd C:\OpenStack\build\openstack\networking-hyperv
-    & pip install C:\OpenStack\build\openstack\networking-hyperv
-    if ($LastExitCode) { Throw "Failed to install networking-hyperv from repo" }
+    Write-Host "Content of $buildDir\neutron"
```
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1]: Where is the content?

File: HyperV/scripts/create-environment.ps1 [hunk 7]
```diff
@@ -202,53 +212,82 @@ function cherry_pick($commit) {
...
+    if ($LastExitCode) { Throw "Failed to install networking-hyperv from repo" }
     popd
 }
 
 ExecRetry {
-    pushd C:\OpenStack\build\openstack\compute-hyperv
-    & pip install C:\OpenStack\build\openstack\compute-hyperv
-    if ($LastExitCode) { Throw "Failed to install Hyperv-Compute fom repo" }
+    if ($isDebug -eq  'yes') {
+        Write-Host "Content of $buildDir\nova"
```
Comment (ociuhandu) [inline HyperV/scripts/create-environment.ps1]: nova or compute-hyperv?

File: HyperV/scripts/gerrit-git-prep.sh [hunk 8]
```diff
@@ -76,16 +81,21 @@ then
     echo "Triggered by: $GERRIT_SITE/$ZUUL_CHANGE"
 fi
 
-if [ -d "$BUILD_DIR" ]
+if [ ! -d "$BUILD_DIR" ]
 then
-  rm -rf "$BUILD_DIR"
+  mkdir -p "$BUILD_DIR"
```
Comment (ociuhandu) [inline HyperV/scripts/gerrit-git-prep.sh:86]: i guess this should be outside "if"

Comment (rbuzatu90) [inline HyperV/scripts/gerrit-git-prep.sh:86]: hmmm why ? if the  $BUILD_DIR doesnt exist we create it

Comment (ociuhandu) [inline HyperV/scripts/gerrit-git-prep.sh:86]: good. :+1:

**Pull request merged**